3.1 Ops: deploying and testing in sepolia

gpmayorga opened the pull request:
This PR

## Wire Adapters
- New logic safeguards to make sure async configurations are wired properly (one adapter existing on chain A but not chain B)
- Update adapter registration methods to use the new guardians

## Cross chain tests
- BaseTest file with common test data functions
- TestData for standalone chain data creation
- simpletest file to send a sigle `.notifyPool` cross chain message (quick and granular adapters test)
- Cross chain files for pools and vaults in a multi-chain configuration (Hub VS Spoke operations). Only for live testnets. (Multiple  Hub & Spoke messages through all configured adapters)


## New deployment options
- Deploy ALL sepolia configured networks 
- Wire all adapters
- Run all tests
- feature: Retry if a step fails
- feature: Save state between runs to be able to resume after an error

## Adapters
- Added adapters.layerZero.endpoint and adapters.layerZero.layerZeroEid across testnets and mainnets.
- Set adapters.layerZero.deploy=true. Removed any delegate from JSON (delegate comes from env).

References: mainnet and testnet endpoints/eids per LayerZero docs:
https://docs.layerzero.network/v2/deployments/deployed-contracts?stages=mainnet
https://docs.layerzero.network/v2/deployments/deployed-contracts?stages=testnet

lemunozm committed:
Move message related files to core/messages (#698)

* move message files

* fix fmt

gpmayorga committed:
First try at recovering #678

lemunozm commented on script/LaunchDeployer.s.sol:
```diff
@@ -74,6 +74,13 @@ contract LaunchDeployer is FullDeployer {
+            address protocolAdminEnv = vm.envAddress("PROTOCOL_ADMIN");
+            require(protocolAdminEnv != address(0), "PROTOCOL_ADMIN not set for LayerZero deployment");
+            console.log("LayerZero delegate (PROTOCOL_ADMIN):", vm.toString(protocolAdminEnv));
+        }
```
NIT. We can consider `PROTOCOL_ADMIN` to always exist, otherwise, the `adminSafe` in `FullInput` will panic trying to fetch it

gpmayorga replied:
removed!

gpmayorga committed:
deployed sepolia and arbitrum sepolia

gpmayorga committed:
take a stab at wire adapters (still failing)

hieronx committed:
Revert on estimate if destination is not setup (#702)

wischli committed:
ci: add ward check script (#696)

* ci: add ward check script

* fix: script relative dir

* ci: move ward coverage to ci.yml as optional check

* ci: make ward coverage check mandatory

* tests: add missing ward coverage

hieronx committed:
Tests: Change `vm.assume` to `bound`s (#705)

* Change vm.assume calls to bounds

* Format

wischli committed:
fix: uninitialized network multiAdapter views

wischli committed:
apply suggestion from code review & fmt

wischli committed:
ref: move _getFirstAdapterDetails to bottom

wischli committed:
chore: update benches

wischli committed:
Merge remote-tracking branch 'origin/main' into sepolia-deployments-3.1

hieronx committed:
Merge branch 'sepolia-deployments-3.1' of https://github.com/centrifuge/protocol into sepolia-deployments-3.1

gpmayorga committed:
All contracts for Sepolia networks deployed and verified

gpmayorga committed:
Ensure salt also applied to root contract

gpmayorga committed:
small python deployment fixes

hieronx committed:
Add cancelation flow to test data (#709)

* improve batch docs (#706)

* Add cancelation flow to test data

* Fix build

---------

Co-authored-by: Luis Enrique Muñoz Martín <lemunozm@gmail.com>
Co-authored-by: Guillermo Perez <gpmayorga@users.noreply.github.com>

gpmayorga committed:
new wireadapters with some safeguards for asymmetrical adapter configs

gpmayorga committed:
correct Eid for layerzero

lemunozm commented on script/WireAdapters.s.sol:
```diff
@@ -4,16 +4,229 @@ pragma solidity 0.8.28;
+    /// @param remoteNetwork Name of the remote network (e.g., "base-sepolia")
+    /// @param remoteConfig JSON config of the remote network
+    /// @param protocolGuardian ProtocolGuardian contract for wiring operations (allows re-wiring)
+    function _wireWormhole(
```
Do you think we can factorize these methods to work for any adapter? Seems most code structure is the same

gpmayorga replied:
Refactor done!

gpmayorga replied:
refactor done!

lemunozm replied:
Looks super simplified now!! 🥇

gpmayorga committed:
Modular test scripts and cross chain tests

lemunozm commented on script/BaseTestData.s.sol:
```diff
@@ -0,0 +1,373 @@
+     * @notice Perform sync vault test flow (simple deposit)
+     * @dev This is the sync vault test flow from TestData.s.sol
+     */
+    function testSyncVaultFlow(PoolId poolId, ShareClassId scId, ERC20 token, uint128 investAmount) internal {
+        IShareToken shareToken = IShareToken(spoke.shareToken(poolId, scId));
+        SyncDepositVault vault = SyncDepositVault(shareToken.vault(address(token)));
+
+        token.approve(address(vault), investAmount);
+        vault.deposit(investAmount, msg.sender);
+    }
```
NOTE: Sync vaults uses async redemptions, which probably need to be tested in this path

gpmayorga replied:
Added https://github.com/centrifuge/protocol/blob/fa792a6e9f0095f3ef3f1feaa82f6e1654d313cf/script/BaseTestData.s.sol#L301-L312
I hope that's the async redemptions you were referring to

lemunozm commented on script/crosschain/TestCrossChainHub.s.sol:
```diff
@@ -0,0 +1,275 @@
+ *   POOL_INDEX_OFFSET - Offset to add to pool indices (default: current timestamp % 1000)
+ *   TEST_RUN_ID - Custom identifier for this test run (used in pool metadata)
+ * 
+ * Usage:
```
Thanks for adding this guide!

gpmayorga committed:
Fix Hub tests

gpmayorga committed:
address PR comment

gpmayorga committed:
Merge branch 'main' into sepolia-deployments-3.1

gpmayorga committed:
Updated testnets deployment

gpmayorga committed:
latest wireadapters

gpmayorga committed:
fix e2e hub-spoke tests for sepolia testnets

gpmayorga committed:
commit missing test shared methods

gpmayorga committed:
Implement ReleaseManager for orchestrated Sepolia testnet deployments and enhance argument validation in deploy script

gpmayorga committed:
Update .gitignore to exclude .env.back and JSON files in the anvil directory

gpmayorga committed:
cleanup and refactor some var names

hieronx committed:
Remove gas buffer percentage in adapters

hieronx committed:
Increase base cost to 50k

gpmayorga committed:
Merge remote-tracking branch 'origin/simplify-adapters' into sepolia-deployments-3.1

gpmayorga committed:
Update adapter configurations and deployment scripts for Sepolia. Removed deprecated adapter addresses, added new ones, and introduced a new script for deploying only messaging adapters. Enhanced deployment validation and added support for deploying adapters separately.

gpmayorga committed:
pseudo-random pool index selection with environment variable override, ensuring valid index range (100-999)

gpmayorga committed:
Merge remote-tracking branch 'origin/main' into sepolia-deployments-3.1

gpmayorga committed:
sepolia & base sepolia redeploy

gpmayorga committed:
deployed arbitrum sepolia

gpmayorga committed:
New deployment features (anvil still buggy)

gpmayorga committed:
git ignore anvil files

gpmayorga committed:
Add cross-chain vault deployment functionality

- Introduced `XcAsyncVaultParams` and `XcSyncVaultParams` structs for cross-chain vault configurations.
- Implemented `deployAsyncVaultXc` and `deploySyncDepositVaultXc` functions to handle cross-chain vault deployments.

gpmayorga commented:
PR ready to merge, a few notes:

- `TestData.s.sol` functions **exactly as before** - same behavior, refactored implementation
- Cross-chain test scripts (`TestCrossChainHub.s.sol`, `TestCrossChainSpoke.s.sol`) are experimental and **not in CI**, thus it's safe to merge them with little to no review for now as they shouldn't impact the audit at all and the fix can be pushed later.

gpmayorga committed:
fixed import order

gpmayorga committed:
Fix imports unused

gpmayorga committed:
fixed import CI check

gpmayorga committed:
fixed CI checks

gpmayorga committed:
Add readme to test folder and move testdata

gpmayorga committed:
Moved test files

gpmayorga committed:
moved test files under /test/

gpmayorga committed:
Revert anvil.py changes - moved to separate PR

gpmayorga committed:
Update import paths

gpmayorga committed:
Fix TestData script path for anvil deployment

hieronx committed:
Merge branch 'sepolia-deployments-3.1' of https://github.com/centrifuge/protocol into sepolia-deployments-3.1

hieronx committed:
Format, benchmarks

hieronx committed:
Fix

hieronx committed:
Move

hieronx committed:
Fix typo

hieronx committed:
Disable sync vault redemption

hieronx committed:
Add whitelisting

hieronx committed:
Fix typo
